test stdin fails sometimes

```
for i in {1..100}; do make c -C build/tests/stdin/; done
```
```
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend *** FAILED out on stdintest.fz
7,8c7,8
< %s😀: %s😀
< %s😀: %s😀
---
> %s😀: ���
> %s😀: ���

make: *** [../simple_inp.mk:50: c] Error 1
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
make: Leaving directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
make: Entering directory '/home/sam/openvscode-server-fuzion/vscode-fuzion/fuzion/build/tests/stdin'
cat stdintest.fz.stdin | ../../bin/check_simple_example c "../../bin/fz " stdintest.fz || exit 1
RUN stdintest.fz using  c  backend PASSED.
....
```

--- a/include/posix.c
+++ b/include/posix.c
@@ -858,7 +858,7 @@ int fzE_pipe_read(int64_t desc, char * buf, size_t nbytes){
 }
 
 
-// return -1 on error, thenumber of written bytes otherwise
+// return -1 on error, the number of written bytes otherwise
 int fzE_pipe_write(int64_t desc, char * buf, size_t nbytes){
   return write((int) desc, buf, nbytes);
 }

--- a/modules/base/src/os/process.fz
+++ b/modules/base/src/os/process.fz
@@ -93,29 +93,6 @@ module:public process(public pid i64, module std_in, std_out, std_err i64, name 
           unit
 
 
-  # write bytes to stdin of child process
-  #
-  public write_bytes (bytes Sequence u8) outcome i32
-  =>
-    write_bytes std_in bytes
-
-
-  # helper function
-  #
-  write_bytes (desc i64, bytes Sequence u8) outcome i32
-  pre debug: bytes.count > 0
-  =>
-    bytes
-      .chunk pipe_buffer_size
-      .reduce_or_error 0 (r, t)->
-        arr := t.as_array
-        bw := fzE_pipe_write desc arr.internal_array.data arr.count
-        if bw = -1
-          abort (outcome i32) (error "writing to $(desc_name desc) of process $(pid). Wrote $r bytes already." fzE_last_error)
-        else
-          r+bw
-
-
   # install buffered reader for reading from stdout of process
   # and run `f`.
   public with_out(T type, LM type : mutate, f () -> outcome T) outcome T =>
@@ -130,12 +107,6 @@ module:public process(public pid i64, module std_in, std_out, std_err i64, name 
 
   public with_in(T type, LM type : mutate, f () -> T) outcome T =>
     (io.buffered LM).Writer.instate _ (std_in_writer LM) f
-
-
-  # write string to stdin of child process
-  #
-  public write_string (s String) outcome i32 =>
-    write_bytes s.utf8
 
 
   # pipe this processes output to new process
@@ -155,8 +126,14 @@ module:public process(public pid i64, module std_in, std_out, std_err i64, name 
         for p1rb := std_out_read_handler.read pipe_buffer_size
         while
           match p1rb
-            s Sequence => (write_bytes p.std_in s).ok
-            * => false
+            s Sequence =>
+              lm : mutate is
+              lm ! ()->
+                _ := p.with_in unit lm ()->
+                  _ := (io.buffered lm).Writer.env.write s
+                true
+            io.end_of_file => false
+            e error => panic e
 
         # NYI: UNDER DEVELOPMENT: error handling
         _ := fzE_pipe_close p.std_in

--- a/tests/process/test_process.fz
+++ b/tests/process/test_process.fz
@@ -45,10 +45,11 @@ test_process =>
 
   test "write to stdin of process" ()->
     (os.process.start "cat").bind p->
-      _ := p.write_string "Hello from the other side."
-      # closing input so cat knows we are done and must write to standard out
-      say p.close_in
       _ := lm ! ()->
+        _ := p.with_in unit lm ()->
+          _ := (io.buffered lm).Writer.env.write "Hello from the other side."
+        # closing input so cat knows we are done and must write to standard out
+        say p.close_in
         _ := p.with_out unit lm ()->
           say ((io.buffered lm).read_string 1E9).or_panic
 

